foundations/javascript_basics/clean_code.md

```diff
@@ -42,63 +47,212 @@ const sumArray = function(array) {
 sumArray([2, 2, 2]);
 ~~~
 
-Believe it or not, both of those functions do the exact same thing \(in the exact same way!\), and both of them are perfectly valid code, but obviously the second one is much easier to follow.  Imagine you're working on a project with someone else and they've written the first function... how long is it going to take you to figure out what's going on there so you can do your work?  Imagine you're working on a project all by yourself and YOU wrote the first function a week or two ago... chances are good that you aren't going to remember exactly what you were up to there and it's _still_ going to take you a good while to figure it all out again.
+Which of these examples do you find easier to read? It's immediately clear that the latter one is more meaningful. Surprisingly, both of these functions perform the exact same task \(in the exact same way!\), and both are valid code. But the second one is much more readable. Why?
+
+In the first example, single-letter variables are used, there is a semicolons in the middle of lines, and the indentation is inconsistent. The result is a piece of code that is confusing and messy. 
+
+Imagine you're collaborating on a project with someone who has written the first function. How long will it take you to decipher what's going on so you can continue with your work? Or perhaps you've written it yourself some time ago and completely forgot that it even existed. In both situations, you will eventually understand what is happening, but it's not going to be fun.
+
+Example B represents clean code. While you may not know  what each part does, it's much easier to guess what is happening because the function and variables are named clearly. The indentation follows a consistent and logical pattern and there are no semicolons interrupting the code within a line.
+
+Single characters can be used as variable names in the context of a loop or a callback function, but avoid them elsewhere.
+
+### Naming functions and variables
+
+In our first example we already touched on the importance of naming things _meaningfully_. Let's break down further what makes a good variable or function name.
+
+#### A good name is descriptive
+
+In our good example the we have a variable `sum`, to which each new `number` from the array is added. The function is named `sumArray` and the function does what the name suggests (if you're thinking "what even is an array", don't worry - you'll learn about them soon). Nice, clean and understandable.
```

Comment:
nit:

I believe the last sentence requires on more comma after clean:

`Nice, clean, and understandable.`

Reply:
Use of [the cereal comma](https://en.m.wikipedia.org/wiki/Serial_comma) is not agreed upon, even within American style punctuation, as TOP has gone with. It varies depending on which style guide is being used.

Reply:
As a non-native, I'm hesitant to give any input to this - all I know that the Oxford comma question can be heated :rofl: